Sources/SwiftFormatRules/AddModifierRewriter.swift

```diff
@@ -39,115 +37,127 @@ private final class AddModifierRewriter: SyntaxRewriter {
+    // token.
     guard let modifiers = node.modifiers else {
-      guard
-        let newDecl = removeFirstTokLeadingTrivia(node: node)
-        as? StructDeclSyntax
-      else { return node }
-      return newDecl.addModifier(modifierKeyword)
+      let nodeWithModifier = node.addModifier(modifierKeyword)
+      return relocateTrivia(in: nodeWithModifier) { $0.modifiers }
     }
     guard modifiers.accessLevelModifier == nil else { return node }
     let newModifiers = modifiers.prepend(modifier: modifierKeyword)
     return node.withModifiers(newModifiers)
   }
 
   override func visit(_ node: TypealiasDeclSyntax) -> DeclSyntax {
+    // Check for modifiers, and, if none, insert the modifier and relocate trivia from the displaced
+    // token.
     guard let modifiers = node.modifiers else {
-      guard
-        let newDecl = removeFirstTokLeadingTrivia(node: node)
-        as? TypealiasDeclSyntax
-      else { return node }
-      return newDecl.addModifier(modifierKeyword)
+      let nodeWithModifier = node.addModifier(modifierKeyword)
+      return relocateTrivia(in: nodeWithModifier) { $0.modifiers }
     }
     guard modifiers.accessLevelModifier == nil else { return node }
     let newModifiers = modifiers.prepend(modifier: modifierKeyword)
     return node.withModifiers(newModifiers)
   }
 
   override func visit(_ node: InitializerDeclSyntax) -> DeclSyntax {
+    // Check for modifiers, and, if none, insert the modifier and relocate trivia from the displaced
+    // token.
     guard let modifiers = node.modifiers else {
-      guard
-        let newDecl = removeFirstTokLeadingTrivia(node: node)
-        as? InitializerDeclSyntax
-      else { return node }
-      return newDecl.addModifier(modifierKeyword)
+      let nodeWithModifier = node.addModifier(modifierKeyword)
+      return relocateTrivia(in: nodeWithModifier) { $0.modifiers }
     }
     guard modifiers.accessLevelModifier == nil else { return node }
     let newModifiers = modifiers.prepend(modifier: modifierKeyword)
     return node.withModifiers(newModifiers)
   }
 
-  func removeFirstTokLeadingTrivia(node: DeclSyntax) -> DeclSyntax {
-    let withoutLeadTrivia = replaceTrivia(
+  /// Moves trivia in the given node to correct the placement of potentially displaced trivia in the
+  /// node after the first modifier was added to the given node. The added modifier is assumed to be
+  /// the first and only modifier of the node. After the first modifier is added to a node, any
+  /// leading trivia on the token immediately after the modifier is considered displaced. This
+  /// method moves that displaced trivia onto the new modifier. When there is no displaced trivia,
+  /// this method does nothing and returns the given node as-is.
+  /// - Parameter node: A node that was updated to include a new modifier.
+  /// - Parameter modifiersProvider: A closure that returns all modifiers for the given node.
+  func relocateTrivia<NodeType: DeclSyntax>(
```

Comment:
Swift naming conventions for methods (which, to be completely fair, are spottily followed through the formatter codebase) typically use verb phrases for `Void`-returning methods and noun phrases that describe the thing being returned for everything else.

So a better name for this method might be `nodeByRelocatingTrivia(in:for:)`, which adds clarity to the call sites about what the output of the method is.

Reply:
Renamed. Thanks for the additional context!